### DISCUSSION TITLE
replace username with mobile number for Register or Login

### DISCUSSION THREAD

**Discussion by khan-umer** [codeigniter4/shield#519] [Q&A] [answered]
Hello everybody,

how can we delete **username** column from **users** table & 
also add **mobile number** field for register & login.

I'm  new in shield so please consider it.
what are the steps to achieve it or any working example will be appreciated.
Thanks.

**Comment by datamweb** [comment, marked as answer]
Hello,
> how can we delete **username** column from **users** table

You should not do this, it will cause a lot of problems.

> add **mobile number** field for register & login.

 If you plan to add a phone column, see  #333 and #430

Also note that currently it is not possible to log in with a custom field, if you think this feature should be added to the shield, please contribute to the discussion #334 .

When I am sure that this possibility is needed by many people, I or any person will send PR.

**Note:**
Another way is to get the user's phone number instead of the username, then register the user and log in by changing the default validations.
I haven't really looked into this, and obviously this method doesn't make sense.

**Reply by khan-umer** [reply]
@datamweb 

> Also note that currently it is not possible to log in with a custom field ....

it seems limitation in shield. we need to provide other login option like **Login with OTP** which is another good option for login.

I agree with @lonnieezell comment in discussion #334 where he list down use cases. 
I hope we will get updates soon about issue #334

**Reply by datamweb** [reply]
> Login with OTP

By default, Shield has email otp.
I think by looking at `src/Controllers/MagicLinkController.php` you can use it for mobile login otp. I don't think it's difficult, but I haven't tried it in practice.

**Reply by khan-umer** [reply]
> By default, Shield has email otp.

 okey I will try.

 thanks @datamweb
